### PR TITLE
Added functionality to Allow/Block devices by MAC and to call the Get…

### PR DESCRIPTION
**Pull request by gskluzacek**
Added functionality to Allow/Block devices by MAC and to call the Get AttachDevice2 SOAP Service

added 4 new methods to the Netgear class
 - config_start
 - config_finish
 - allow_block_device - arguments of mac_addr, device_status ('Block' or 'Allow')
   calls: config_start() & config_finish() methods
 - get_attached_devices_2 - list of attached devices that has up to 15 attributes
   this method has a dependency on untangle to parse returned XML for attached devices

added 1 new attribute on the Netgear class
 - self.config_started, default to False - tracks state of config_start() & config_finish() methods

Added 4 new SOAP action URNs, 1 for each of the new methods above
 - ACTION_CONFIG_START :: DeviceConfig:1#ConfigurationStarted
 - ACTION_CONFIG_FINISH :: DeviceConfig:1#ConfigurationFinished
 - ACTION_ALLOW_BLOCK_DEVICE :: DeviceConfig:1#SetBlockDeviceByMAC
 - ACTION_GET_ATTACHED_DEVICES_2 :: DeviceInfo:1#GetAttachDevice2

Added 4 new SOAP requests, 1 for each of the new methods above
 - SOAP_CONFIG_START :: takes session_id as argument
 - SOAP_CONFIG_FINISH :: takes session_id as argument
 - SOAP_ALLOW_BLOCK_DEVICE :: takes session_id, mac_addr, device_status as argument
 - SOAP_ATTACHED_DEVICES_2 :: takes session_id as argument

### CODE REVIEW

File: pynetgear/__init__.py [hunk 1]
```diff
@@ -160,6 +161,85 @@ def tofloats(lst): return (float(t) for t in lst)
...
+        """
+        Allow or Block a device via its Mac Address.
+
+        Pass in the mac address for the device that you want to set. Pass in the 
+        device_status you wish to set the device to: Allow (allow device to access the
+        network) or Block (block the device from accessing the network).
+        """
+        _LOGGER.info("Allow block device")
+        if self.config_started
+        	_LOGGER.error("Inconsistant configuraiton state, configuraiton already started")
```
Comment (KKDad) [inline pynetgear/__init__.py]: Typo

File: pynetgear/__init__.py [hunk 2]
```diff
@@ -160,6 +161,85 @@ def tofloats(lst): return (float(t) for t in lst)
...
+        	return False
+        message = SOAP_ALLOW_BLOCK_DEVICE.format(session_id=SESSION_ID, device_status=device_status, mac_addr=mac_addr)
+        success, _ = self._make_request(
+            ACTION_ALLOW_BLOCK_DEVICE, message)
+        if not success:
+        	_LOGGER.error("Could not successfully call allow/block device")
+        	return False
+        	
+        if not self.config_finish():
+        	_LOGGER.error("Inconsistant configuraiton state, configuraiton already finished")
```
Comment (KKDad) [inline pynetgear/__init__.py]: Same typo 2 more times :-)

**Comment by MatMaul** [comment]
And complete the README with the new public methods please.

**Comment by MatMaul** [comment]
Modified version commited in 22ada8fe90227973dd2e349064d8253f19fc60c9.

**Pull request closed without merging**